content/client-lib-development-guide/features.textile

```diff
@@ -599,6 +599,16 @@ h3(#realtime-channel). Channel
 ** @(RTL7d)@ Messages delivered are automatically decoded based on the @encoding@ attribute; see REST @Channel@ encoding features. Tests should exist to publish and subscribe to encoded messages using the "AES 128":https://github.com/ably/ably-common/blob/main/test-resources/crypto-data-128.json and "AES 256":https://github.com/ably/ably-common/blob/main/test-resources/crypto-data-256.json fixture test data
 ** @(RTL7e)@ If a message cannot be decoded or decrypted successfully, it should be delivered to the listener with the @encoding@ attribute set indicating the residual encoding state, and an error should be logged
 ** @(RTL7f)@ A test should exist ensuring published messages are not echoed back to the subscriber when @echoMessages@ is set to false in the @Realtime@ library constructor
+** @(RTL7g)@ Subscribe with a dictionary argument subscribes a listener which filters which messages to accept based on the dictionary values
+*** @(RTL7g1)@ An empty dictionary functions identically to "RTL7a":#RTL7a
+*** @(RTL7g2)@ A dictionary with multiple key/value pairs should require each entry to be satisfied in order to accept the Message
+*** @(RTL7g3)@ Keys currently must be one of @ref@ or @type@ or a negated version of these (@-ref@ and @-type@)
+**** @(RTL7g3a)@ @ref@ and @type@ refer to @ref.timeserial@ and @ref.type@ in the Message body respectively
```

Comment:
Had a chat with Paddy this morning, we're going to put the ref and timeserial fields in extras rather than the main message body.

This is because it allows SDKs that we haven't added official support in to still see and use these features.